tests/iter.rs

```diff
@@ -21,10 +21,10 @@ fn insufficient_space() {
     assert_eq!(meta.count, 3);
 
     // Ensure that we can't iterate if we make a state vector that is too small.
-    assert!(Settings::iter_paths::<1, 256>().is_err());
+    assert!(Settings::iter_paths::<1, String<256>>().is_err());
 
     // Ensure that we can't iterate if the topic buffer is too small.
-    assert!(Settings::iter_paths::<10, 1>().is_err());
+    // assert!(Settings::iter_paths::<10, String<1>>().is_err());
```

Comment:
```suggestion
    assert!(Settings::iter_paths::<10, String<1>>().is_err());
```

Reply:
Or is the problem that we no longer detect this as an error? That seems like a problem

Reply:
See the PR description.
I think it's ok to skip it as we've always found it necessary to do a similar check and redundant check elsewhere as well.

Reply:
Oh sorry, I thought this was related to the first iteration (next()) not returning an error. The construction not returning an error is fine. Can we remove the broken test?